crates/uv-client/src/base_client.rs

```diff
@@ -198,36 +246,60 @@ impl<'a> BaseClientBuilder<'a> {
 
                 client.build()
             }
-            Connectivity::Offline => reqwest_middleware::ClientBuilder::new(client.clone())
+            Connectivity::Offline => reqwest_middleware::ClientBuilder::new(client)
                 .with(OfflineMiddleware)
                 .build(),
-        };
-
-        BaseClient {
-            connectivity: self.connectivity,
-            client,
-            timeout,
         }
     }
 }
 
 /// A base client for HTTP requests
 #[derive(Debug, Clone)]
 pub struct BaseClient {
-    /// The underlying HTTP client.
+    /// The underlying HTTP client that enforces valid certificates.
     client: ClientWithMiddleware,
+    /// The underlying HTTP client that accepts invalid certificates.
+    dangerous_client: ClientWithMiddleware,
     /// The connectivity mode to use.
     connectivity: Connectivity,
     /// Configured client timeout, in seconds.
     timeout: u64,
+    /// The host that is trusted to use the insecure client.
+    trusted_host: Vec<TrustedHost>,
+}
+
+#[derive(Debug, Clone, Copy)]
+enum Security {
+    /// The client should use secure settings, i.e., valid certificates.
+    Secure,
+    /// The client should use insecure settings, i.e., skip certificate validation.
+    Insecure,
 }
 
 impl BaseClient {
-    /// The underlying [`ClientWithMiddleware`].
+    /// The underlying [`ClientWithMiddleware`] for secure requests.
     pub fn client(&self) -> ClientWithMiddleware {
         self.client.clone()
     }
 
+    /// Selects the appropriate client based on the host's trustworthiness.
+    pub fn for_host(&self, url: &Url) -> &ClientWithMiddleware {
+        if self
+            .trusted_host
+            .iter()
+            .any(|trusted_host| match trusted_host {
+                TrustedHost::Host(host) => url.host_str() == Some(host.as_str()),
+                TrustedHost::HostPort(host, port) => {
+                    url.host_str() == Some(host.as_str()) && url.port() == Some(*port)
+                }
+            })
+        {
+            &self.dangerous_client
```

Comment:
The only way to access this client is via `for_host`. It's not exposed anywhere else.